### PR TITLE
Add an Approval toggle button to frontend problem page.

### PR DESCRIPTION
**Pull request by zpChris**
### List of Changes:
- Add an "Approval" toggle button to the frontend for the problem edit (not create) page.

### Notes:
- Closed #135 and created this PR because the "Files changed" weren't updating properly on #135.
- This is not fully implemented, as I could not figure out how to connect the toggle button to the `handleChange` function; whenever I tried to connect this, the code through errors and wouldn't compile, so I'm looking for help on this @alankbi?
- After help on the above, I think this can be a helpful PR for @sean-xcamp to learn how different components on the frontend interact.
- I added a new text size to `Theme.tsx`; I'm open to removing this, but the "Approval ON" text didn't look great with any of the existing options, so I thought I'd just add a new size.
- I tried to make this design comparable to the slider, but it may not be exactly "sized" the same, so adjustments may need to be made once it's added to the `Room.tsx` page. Eventually, the "disabled" styling should likely also be implemented.

### Screencast and Images:

[Toggle Button on Problem Page](https://www.loom.com/share/ee955158b6ae483baa4c41c620641220)

**Toggle Button Design (OFF)**
Toggle Button Design (OFF)

**Toggle Button Design (ON)**
Toggle Button Design (ON)

### CODE REVIEW

**Comment by zpChris** [comment]
[Toggle button for problem approval](https://trello.com/c/IeZyQITt/323-toggle-button-for-problem-approval)

**Review by alankbi — COMMENTED** [review]
Left a few nits, and also here are some more general suggestions/ideas:

* On the all problems page, could we maybe replace the problem id with the approval status? The problem id isn't too useful to see and it could be nice to have the approval status there so we know easily what problems still need to be approved. 
* For the toggle button, could we change the default background color to some kind of gray? Similar to the iOS toggle buttons
* On this branch, any time I try to save the problem (without making any changes), it gives the error "Please enter a value for each required field." This might be the error you mentioned in the PR description though

> This is not fully implemented, as I could not figure out how to connect the toggle button to the handleChange function; whenever I tried to connect this, the code through errors and wouldn't compile, so I'm looking for help on this @alankbi?

Sure, I can take a look at this (assuming you haven't already fixed it in the meantime? It's taken me quite a while to get to this PR lol)

**Comment by alankbi** [comment]
> This is not fully implemented, as I could not figure out how to connect the toggle button to the handleChange function; whenever I tried to connect this, the code through errors and wouldn't compile, so I'm looking for help on this @alankbi?

Just pushed a commit to address this! I think it should work, but I didn't do any rigorous testing so feel free to test it out/change it if there are any bugs

File: frontend/src/components/core/Button.tsx [hunk 1]
```diff
@@ -204,3 +204,56 @@ export const InlineRefreshIcon = styled.i.attrs(() => ({
...
+
+export const ToggleButtonInput = styled.input.attrs((props: CheckboxProps) => ({
+  type: 'checkbox',
+  checked: props.checked,
+}))<CheckboxProps>``;
+
+export const ToggleButtonSpan = styled.span.attrs((props: CheckboxProps) => ({
+  style: {
+    backgroundColor: props.checked ? `${ThemeConfig.colors.sliderBlue}` : `${ThemeConfig.colors.white}`,
+  },
```
[the remark this answers is not in the document]
Comment (zpChris) [reply frontend/src/components/core/Button.tsx:229]: No reason - made this change 👍

File: frontend/src/components/core/Button.tsx [hunk 2]
```diff
@@ -204,3 +204,56 @@ export const InlineRefreshIcon = styled.i.attrs(() => ({
     box-shadow: 0 2px 12px rgba(0, 0, 0, 0.24);
   }
 `;
+
+export const ToggleButtonLabel = styled.label`
+  position: relative;
+  display: inline-block;
+  width: 55px;
```
[the remark this answers is not in the document]
Comment (zpChris) [reply frontend/src/components/core/Button.tsx:211]: Sure, I'll create a new `tsx` file for these toggle button components!

File: frontend/src/components/problem/ProblemDisplay.tsx [hunk 3]
```diff
@@ -64,6 +67,25 @@ const SettingsContainer = styled.div`
...
 
+type ShowProps = {
+  show: boolean,
+};
+
+const ApprovalContainer = styled.div.attrs((props: ShowProps) => ({
+  style: {
+    display: props.show ? 'inline-block' : 'none',
+  },
+}))<ShowProps>`
```
[the remark this answers is not in the document]
Comment (zpChris) [reply frontend/src/components/problem/ProblemDisplay.tsx:78]: Done!

File: frontend/src/components/problem/ProblemDisplay.tsx [hunk 4]
```diff
@@ -205,6 +228,15 @@ function ProblemDisplay(props: ProblemDisplayParams) {
   // Handle description change
   const handleDescriptionChange = (value: string) => handleChange({ target: { name: 'description', value } });
 
+  // Toggle the problem approval status
+  const toggleProblemApproval = () => {
+    if (problemApproval) {
+      setProblemApproval(false);
+    } else {
+      setProblemApproval(true);
+    }
```
[the remark this answers is not in the document]
Comment (zpChris) [reply frontend/src/components/problem/ProblemDisplay.tsx]: I think this is outdated since your latest change - thanks for addressing this 👍

**Review by alankbi — APPROVED** [review]
Looks good!

**Pull request merged**